### ISSUE TITLE
Syntax highlighting breaks with complicated expression in string literal in C#

### ISSUE THREAD

**Issue by hughbe** [sublimehq/Packages#646]
If you look at the following code in the latest version of sublime text 3, you see that the syntax colorer highlights everything orange after the end of the string literal (see screenshot) incorrectly:
![30ed94ff84d0114715bb9e528eeffb3716a35f7b](https://cloud.githubusercontent.com/assets/1275900/18953541/e1cfc308-8646-11e6-9207-27deb8998a1c.png)

```
namespace Namespace 
{
    class C
    {
            public override string ToString() =>
            string.Format(CultureInfo.InvariantCulture, "catch({0}) [{1}->{2}]", ExceptionType.Name, HandlerStartIndex, HandlerEndIndex);
    }
}
```

**Comment by keith-hall** [comment]
it's not related to the string literal, it's because the current C# syntax definition doesn't support lambdas. This is being addressed in #93.

**Comment by wbond** [comment]
The rewrite from #93 just landed in master, and this issue has been resolved
